src/main/java/net/minecraftforge/event/entity/living/PotionEvent.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2017.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package net.minecraftforge.event.entity.living;
+
+import javax.annotation.Nullable;
+import javax.annotation.Nonnull;
+
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.potion.Potion;
+import net.minecraft.potion.PotionEffect;
+
+import net.minecraftforge.fml.common.eventhandler.Cancelable;
+import net.minecraftforge.fml.common.eventhandler.Event.HasResult;
+
+/**
+ * This Event and its subevents gets fired from  {@link EntityLivingBase} on the  {@link MinecraftForge#EVENT_BUS}.<br>
+ */
+public class PotionEvent extends LivingEvent
+{
+    @Nonnull
+    protected final PotionEffect effect;
+    
+    public PotionEvent(EntityLivingBase living, PotionEffect eff)
+    {
+        super(living);
+        effect = eff;
+    }
+    /**
+     * Retuns the PotionEffect
+     */
+    @Nonnull
+    public PotionEffect getPotionEffect()
+    {
+        return effect;
+    }
+    
+    /**
+     * This Event is fired when a Potion is about to get removed from an Entity
+     * This Event is {@link Cancelable}
+     * This Event does not have a result.
+     */
+    @Cancelable
+    public static class PotionRemoveEvent extends PotionEvent
+    {
+        private final Potion toRemove;
+        
+        public PotionRemoveEvent(EntityLivingBase living, Potion  potion)
+        {
+            super(living, living.getActivePotionMap().get(potion));
+            toRemove = potion;
+        }
+        
+        /**
+         * Retuns the Potion wich is tried to remove from the Entity
+         */
+        public Potion getPotion()
+        {
+            return toRemove;
```

Comment:
`getPotion()` should return a field named `potion`. `toRemove` is not a good name here, it should match.